src/main/java/org/elasticsearch/index/query/FutureBaseQueryParser.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+
+import java.io.IOException;
+
+/**
+ * Class used during the query parsers refactoring.
+ * All query parsers that have a refactored "toXContent" method can be changed to extend this instead of {@link BaseQueryParser}.
+ * Keeps old {@link QueryParser#parse(QueryParseContext)} method as a stub delegating to
+ * {@link QueryParser#fromXContent(QueryParseContext)} and {@link QueryBuilder#toQuery(QueryParseContext)}}
+ */
+public abstract class FutureBaseQueryParser implements QueryParser {
```

Comment:
about the name, nitpick: given that we are now changing all parsers to extend BaseQueryParser, then we are moving each one over to FutureBaseQueryParser, which then will need to be renamed to something more human (BaseQueryParser again once the current BaseQueryParser is gone), and at some point will be removed too.... should we use the name that we want for it right now and keep it till it gets deleted? That would help causing less noise in all the query parsers I think. I would probably call this one BaseQueryParser and the other one BaseQueryParserTemp or whatever, that one is the first one that goes away anyway. Makes sense?